### PR TITLE
send index with button click event

### PR DESCRIPTION
**Pull request by koush**
Easier to know which button was clicked, vs doing a text comparison.

### CODE REVIEW

**Comment by koush** [comment]
Crap. I meant to send these are separate pull requests, but added my subsequent commits.

**Comment by bayleedev** [comment]
This is great! I'll review it when I get home :D

File: lib/notifier.js [hunk 1]
```diff
@@ -53,6 +58,10 @@ class Notifier {
             notificationWindow.close()
         }, 4000)
 
+        const self = electron.remote && electron.remote.BrowserWindow.getCurrentWindow();
+        if (self)
+          self.on('close', notificationWindow.removeAllListeners.bind(notificationWindow));
```
Comment (bayleedev) [inline lib/notifier.js]: I'm not sure why we need this.

It feels similar to:

``` js
notificationWindow.on('close', () => {
  notificationWindow.removeAllListeners()
})
```

but even then, won't the events unbind themselves?

File: lib/notifier.js [hunk 2]
```diff
@@ -53,6 +58,10 @@ class Notifier {
             notificationWindow.close()
         }, 4000)
 
+        const self = electron.remote && electron.remote.BrowserWindow.getCurrentWindow();
```
Comment (bayleedev) [inline lib/notifier.js]: This is throwing an error, it's `electron.remote.getCurrentWindow()` now I believe.

**Comment by bayleedev** [comment]
This works great except that last commit. Maybe you could go more into detail about that one?

I added a [playbook](https://github.com/blainesch/electron-notifications#playbook) in the app so you can test your changes live.

File: lib/notifier.js (hunk 1 above)
Comment (koush) [reply lib/notifier.js]: There's more fixes around this. Will leave info in thread.

**Comment by koush** [comment]
Suppese you have a BrowserWindow that creates a notificationWindow. Ie, not created from the main process. If that BrowserWindow closes _before_ the notificationWIndow, Electron will crash. Attempt to RPC onto a dead object.

So if the _owner/caller_ of the notification is closed, the notification should purge all listeners. The notificationWindow removing all listeners on itself when the notification closes does not address this bug.

**Comment by koush** [comment]
Another alternate solution is to have the caller manage the callbacks on the notificationWindow.

The reason I chose to do it this way, is it's way less of a headach as there is no callback management code.

If there's a better/idiomatic way to do this, please let me know! I'm new to Electron :)

File: lib/notifier.js (hunk 2 above)
Comment (koush) [reply lib/notifier.js]: Fixed

**Comment by bayleedev** [comment]
Oh, that makes sense! I'll test that case manually today!

**Comment by koush** [comment]
Unrelated, but I'm using your project for this. Run Chrome apps without Chrome:
https://github.com/koush/electron-chrome

**Comment by bayleedev** [comment]
I rebased your pr in, still comitted as you.
https://github.com/blainesch/electron-notifications/commits/master?author=koush

**Pull request closed without merging**